### PR TITLE
janitor: add code to clean the DO account for capdo

### PR DESCRIPTION
**Pull request by cpanato**
**What this PR does / why we need it**:
Add go code to clean up resources in the CAPDO DO Account that is older than 2 days to avoid account limits and spend $ for nothing.

I will add a periodic job in prow to run every day

/kind feature

/assign @MorrisLaw @prksu @xmudrii 

mention @timoreimann as well to be aware :)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
janitor: add code to clean the DO account for capdo
```

### CODE REVIEW

File: hack/cleanup/resource_mgt.go
Comment (timoreimann) [inline hack/cleanup/resource_mgt.go:62]: The way godo works is that it [always returns a non-nil error if the response code is > 300](https://github.com/digitalocean/godo/blob/05dc4d920aacfcded4808de4d5b0d7bb78793606/godo.go#L436-L451). Unless we care about a specific 2xx code here, I think we could also drop this status code check.

File: hack/cleanup/resource_mgt.go
Comment (timoreimann) [inline hack/cleanup/resource_mgt.go:54]: nit: this could also be

```go
days := time.Since(dropletCreated).Hours() / 24
```

making `today` unnecessary.

**Review by timoreimann — APPROVED** [review]
Nice work, thanks @cpanato. 👏 

(I have no approval powers here but thought I could give my 2 cents anyway. :) )

File: hack/cleanup/resource_mgt.go
Comment (cpanato) [reply hack/cleanup/resource_mgt.go:62]: dropped

File: hack/cleanup/resource_mgt.go
Comment (cpanato) [reply hack/cleanup/resource_mgt.go:54]: done, thanks for your review!

**Review by prksu — COMMENTED** [review]
@cpanato i think, it would be better if we not depend on anything from `test/e2e`

**Comment by cpanato** [comment]
@prksu should we extract that to some utils package? and then both uses that?

File: hack/cleanup/do-janitor.go
Comment (xmudrii) [inline hack/cleanup/do-janitor.go:50]: Can we make this a const?

File: hack/cleanup/do-janitor.go
Comment (xmudrii) [inline hack/cleanup/do-janitor.go:30]: Why did we remove this function? I'm not sure what exactly it does and did it worked before, but as far as I understand, it would clean leftover resources after the test is done.

This might not be reliable in all cases, for example, if we force push to a branch, Prow might interrupt tests instantly, and that function wouldn't have a chance to finish and clean up resource. But in other cases, it still might be useful, e.g. some failed test or bug.

With Janitor, we would only clean up resources that are 2 days old, every 12 hours, so it might happen that tests leak some resources and that we wait 2 days to remove them, while this function might eventually remove those resources right after tests are finished.

I'm not sure did I understood this correctly, so please correct me if I'm wrong.

Comment (cpanato) [reply hack/cleanup/do-janitor.go:30]: discussed with @prksu and he suggests to remove from the e2e and use the janitor to deal with that.

we can reduce the time, instead of 2 days we can set to 12 hours or even less.

Comment (xmudrii) [reply hack/cleanup/do-janitor.go:30]: @cpanato Thank you for the explanation! I think reducing the time makes sense. 6-12 hours should be enough, IMO.

File: hack/cleanup/do-janitor.go
Comment (cpanato) [reply hack/cleanup/do-janitor.go]: done

File: hack/cleanup/do-janitor.go
Comment (cpanato) [reply hack/cleanup/do-janitor.go]: ok
I also reverted the e2e clean up, that makes sense to keep

Comment (cpanato) [reply hack/cleanup/do-janitor.go]: @xmudrii done, thanks for your review

File: hack/cleanup/do-janitor.go
Comment (prksu) [inline hack/cleanup/do-janitor.go:1]: can we make this under `hack/do-janitor/` directory? to make a bit organized directory
if you don't think so, just ignore this comment

**Review by prksu — COMMENTED** [review]
left small comment
otherwise lgtm

/lgtm
/hold
cc @xmudrii  for final review

File: hack/cleanup/do-janitor.go
Comment (cpanato) [reply hack/cleanup/do-janitor.go]: done @prksu

File: hack/do-janitor/do-janitor.go
Comment (xmudrii) [inline hack/do-janitor/do-janitor.go:52]: Tiny nit: we should rename that variable to hours.

File: hack/do-janitor/do-janitor.go
Comment (xmudrii) [inline hack/do-janitor/do-janitor.go:76]: Same here.

**Review by xmudrii — APPROVED** [review]
Two tiny nits, but you can ignore those. Feel free to remove hold if you think it's ready.
/lgtm

File: hack/do-janitor/do-janitor.go
Comment (cpanato) [reply hack/do-janitor/do-janitor.go]: 🤦 
done PTAL

File: hack/do-janitor/do-janitor.go
Comment (xmudrii) [inline hack/do-janitor/do-janitor.go:75]: We should not divide with 24 here.

```suggestion
		hours := time.Since(lbCreated).Hours()
```

**Review by xmudrii — CHANGES_REQUESTED** [review]
One more comment.

File: hack/do-janitor/do-janitor.go
Comment (cpanato) [reply hack/do-janitor/do-janitor.go:75]: today is not my day :)

Comment (cpanato) [reply hack/do-janitor/do-janitor.go:75]: done

**Review by xmudrii — APPROVED** [review]
Looks great! Thank you for implementing this and addressing all the comments!

/lgtm

**Pull request merged**
